src/constructs/loadbalancing/elb.test.ts

```diff
@@ -14,43 +15,31 @@ describe("The GuClassicLoadBalancer class", () => {
     privateSubnetIds: [""],
   });
 
-  test("overrides the id with the overrideId prop", () => {
-    const stack = simpleGuStackForTesting();
-    new GuClassicLoadBalancer(stack, "ClassicLoadBalancer", { vpc, overrideId: true });
-
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).toContain("ClassicLoadBalancer");
-  });
-
   test("has an auto-generated ID by default", () => {
     const stack = simpleGuStackForTesting();
     new GuClassicLoadBalancer(stack, "ClassicLoadBalancer", { vpc });
 
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).not.toContain("ClassicLoadBalancer");
+    expect(stack).toHaveResourceOfTypeAndLogicalId("AWS::ElasticLoadBalancing::LoadBalancer", /ClassicLoadBalancer.+/);
   });
 
-  test("overrides the id if the stack migrated value is true", () => {
+  test("overrides the logicalId when existingLogicalId is set in a migrating stack", () => {
     const stack = simpleGuStackForTesting({ migratedFromCloudFormation: true });
-    new GuClassicLoadBalancer(stack, "ClassicLoadBalancer", { vpc });
+    new GuClassicLoadBalancer(stack, "ClassicLoadBalancer", { vpc, existingLogicalId: "ClassicLB" });
 
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).toContain("ClassicLoadBalancer");
+    expect(stack).toHaveResourceOfTypeAndLogicalId("AWS::ElasticLoadBalancing::LoadBalancer", "ClassicLB");
   });
 
-  test("does not override the id if the stack migrated value is true but the override id value is false", () => {
+  test("has an auto-generated logicalId by default", () => {
     const stack = simpleGuStackForTesting({ migratedFromCloudFormation: true });
```

Comment:
Minor: we could remove the `{ migratedFromCloudFormation: true }` here just to keep things nice and simple